### PR TITLE
narrow conditions when WARNING_NEUROMORPHO_SOMA_NON_CONFORM is thrown

### PR DESCRIPTION
**Pull request by asanin-epfl**
Solution to #273

### CODE REVIEW

File: src/readers/morphologySWC.cpp [hunk 1]
```diff
@@ -206,6 +206,14 @@ class SWCBuilder
         const Sample& child1 = _children[0];
         const Sample& child2 = _children[1];
```
Comment (tomdele) [inline src/readers/morphologySWC.cpp:208]: I think you should explain this and put a link to the issue

File: src/readers/morphologySWC.cpp [hunk 2]
```diff
@@ -206,6 +206,14 @@ class SWCBuilder
...
 
+        bool isSuited = std::fabs(child1.diameter - d) < morphio::epsilon &&
+                        std::fabs(child2.diameter - d) < morphio::epsilon &&
+                        std::fabs(child1.point[0] - x) < morphio::epsilon &&
+                        std::fabs(child2.point[0] - x) < morphio::epsilon &&
+                        std::fabs(child1.point[2] - z) < morphio::epsilon &&
+                        std::fabs(child2.point[2] - z) < morphio::epsilon;
+        if (!isSuited)
```
Comment (tomdele) [inline src/readers/morphologySWC.cpp:215]: Same you should explain why you don t throw

File: src/readers/morphologySWC.cpp (hunk 1 above)
Comment (asanin-epfl) [reply src/readers/morphologySWC.cpp:203]: ok

File: src/readers/morphologySWC.cpp (hunk 2 above)
Comment (asanin-epfl) [reply src/readers/morphologySWC.cpp:212]: ok

**Comment by tomdele** [comment]
Thanks for the extra explanations/comment in code. I think this is important for MorphIO since we are all in a process of learning about the morphologies and the different formats. Having this in the code makes things much easier.

**Review by tomdele — APPROVED** [review]
I am good with the changes

**Comment by asanin-epfl** [comment]
Thank you @tomdele. We still need to figure out what bifurcation point is, and I need to make some nice images here. The PR should not be reviewed for now

**Comment by tomdele** [comment]
Ok np I removed my review. Let me know when I should redo it.

File: doc/source/specification.rst [hunk 3]
```diff
@@ -23,32 +23,65 @@ does not make any sense in any of the other soma formats.
...
 <http://mathworld.wolfram.com/ConicalFrustum.html>`_. Each point describes coordinates and radius
 of a disk. Each frustums is formed by 2 consecutive disks.
 
+Soma bifurcating
+~~~~~~~~~~~~~~~~
+.. image:: images/swc/split.svg
+   :scale: 100 %
+   :alt: soma two point
+
+An exceptional case of allowed bifurcating soma. TODO this needs clarifying on when such is possible.
```
Comment (asanin-epfl) [inline doc/source/specification.rst:83]: @mgeplf as mentioned in #273, we need to establish when bifurcatings are allowed.

File: doc/source/specification.rst [hunk 4]
```diff
@@ -23,32 +23,65 @@ does not make any sense in any of the other soma formats.
...
-`SOMA_NEUROMORPHO_THREE_POINT_CYLINDERS <https://github.com/BlueBrain/MorphIO/blob/a60b52dfe403ef289455ee2221c1b4fce6418978/src/plugin/morphologySWC.cpp#L206>`_
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-SWC files with a soma made of 3 points will have a soma of type 3-points soma. A 3-points soma is
+Soma single point
+~~~~~~~~~~~~~~~~~
+.. image:: images/swc/single.svg
+   :scale: 100 %
+   :alt: soma single point
+
+A single point sphere soma is interpreted as a cylinder of equivalent area.
```
Comment (tomdele) [inline doc/source/specification.rst:42]: I think we should be more precise about the "point". This is actually not a point but a sphere/circle --> point + a diameter. 
The "point" definition differs from the single point and the two point soma.

In the single point this is a sphere. In the 2 point soma these are 2 circles.
> two points as centers of the cylinder's bases

Comment (tomdele) [reply doc/source/specification.rst:42]: Am I correct ?

Comment (asanin-epfl) [reply doc/source/specification.rst:42]: I will adjust. In my opinion the whole docs require some serious cleaning. I created a ticket for that #277.

Comment (asanin-epfl) [reply doc/source/specification.rst]: Feel free to fix my definitions. I find it hard to adjust only SWC parts without affecting H5 and ASC. In my opinion we need a good general definition, that will be further concretized for each format. Currently MorphIO does not give general definition.

**Comment by tomdele** [comment]
Thanks this is much clearer now

File: doc/source/specification.rst (hunk 4 above)
Comment (tomdele) [reply doc/source/specification.rst]: I think this should be done in a much bigger review about docs. But this takes time and this is not the priority yet. Your current definition is great atm.

**Review by tomdele — APPROVED** [review]

**Comment by asanin-epfl** [comment]
Thank you!

**Pull request merged**